File: interface/data/shipments/ParcelStatus.ts
// Copyright © Spatial Corporation. All rights reserved.

/**
 * Indicates the current status of a {@link Parcel}.
 */
export type ParcelStatus = "Processed" | "Shipped" | "Moving" | "Delivered";

/**
 * An iterable array of parcel statuses.
 */
export const PARCEL_STATUSES: ParcelStatus[] = ["Processed", "Shipped", "Moving", "Delivered"];

File: cloud/Server/Data/Shipments/ParcelStatus.cs
// Copyright © Spatial Corporation. All rights reserved.

namespace Spatial.Cloud.Data.Shipments;

/// <summary>
/// Indicates the current status of a <see cref="Parcel"/>.
/// </summary>
public enum ParcelStatus
{
    /// <summary>
    /// The <see cref="Parcel"/> was processed.
    /// </summary>
    Processed,

    /// <summary>
    /// The <see cref="Parcel"/> was shipped.
    /// </summary>
    Shipped,

    /// <summary>
    /// The <see cref="Parcel"/> is in transit.
    /// </summary>
    Moving,

    /// <summary>
    /// The <see cref="Parcel"/> was delivered.
    /// </summary>
    Delivered
}